### DISCUSSION TITLE
Canon RAW - CR2 convert speed?

### DISCUSSION THREAD

**Discussion by 71m383nd3r** [ImageMagick/ImageMagick#4875] [General] [answered]
Hi,
I have a question what is the fastest format to convert from CR2 to DNG, JPG, PNG, TIFF or some other format?
tnx

**Comment by fmw42** [comment]
Why don't you just do some time tests?  My guess would be uncompressed TIFF.

**Reply by 71m383nd3r** [reply]
I'm on OSX I don't know how to test it with (time) speed calc.
Any CMD for it?
Can I do multiple output at once?

**Reply by fmw42** [reply]
In a terminal window, type

time magick image.cr2 -compress none image.tiff

for IM 7 uncompressed tiff output.  Use whatever suffix you use for cr2 in case .cr2 is wrong or use CR2:image.suffix.

time is a unix command that you can run in your terminal.  If on IM 6, use convert in place of magick.

Please always identify your IM version and platform, when asking questions.

**Reply by 71m383nd3r** [reply]
Version: ImageMagick 7.1.0-24 Q16-HDRI x86_64 2022-02-12
OS: OSX 10.13.6

magick convert MK3_2153.CR2 MK3_2153.tif  48.56s user 2.30s system 90% cpu 55.946 total
magick convert MK3_2153.CR2 MK3_2153.jpg  48.49s user 1.76s system 93% cpu 53.693 total
magick convert MK3_2153.CR2 MK3_2153.png  60.97s user 2.57s system 91% cpu 1:09.50 total

is there a way to make it faster?

**Reply by fmw42** [reply]
You did not use -compress none for tif. So it is taking time to compress it.  There is no acceleration that I am aware. You can reduce colors on PNG and also try different compression quality values. See the docs. You can do -strip to remove all meta data. You can use Q8 Imagemagick.

**Reply by 71m383nd3r** [reply]
with and without (command "-compress none") the TIF file is the same size, and similar time ±2sec.

**Comment by snibgo** [comment, marked as answer]
> I have a question what is the fastest format to convert from CR2 to DNG, JPG, PNG, TIFF or some other format?

I don't understand the question. IM can't write DNG files. IM can read CR2 files by delegating to libraw or dcraw or ufraw-batch. I doubt there is much performance difference between those delegates.

PNG is usually a poor choice for photographs, partly for performance reasons. PNG was not designed for photographs. JPG is usually fast, but comparatively poor quality. TIFF can use various compression methods, or no compression at all.

**Reply by 71m383nd3r** [reply]
Version: ImageMagick 7.1.0-24 Q16-HDRI x86_64 2022-02-12
OS: OSX 10.13.6

tnx, for the info.
What is the fastest TIF option in general?

**Reply by fmw42** [reply]
I do not know. Try each.  My guess would be uncompressed.

**Reply by fmw42** [reply]
compare to -compress none, -compress lzw and -compress zip

**Reply by 71m383nd3r** [reply]
I did 
≈50sec - none compression
≈1m 8sec - LZW
≈1m 10sec - ZIP

does magick use CPU or GPU in general? is the size of the RAM important in general? (for a file of 200MB I don't think RAM is of big value for speed or?

**Reply by fmw42** [reply]
CPU. There there is no GPU acceleration from most delegates.  RAM is important only if you do not have enough. You would know, if it fails.

**Reply by 71m383nd3r** [reply]
Thank you for your time.
Any chance that we will see a GPU acceleration in near future?

**Reply by fmw42** [reply]
I doubt it, since Imagemagick is using delegates for input and output reading and writing.
